lib/actions.ex

```diff
@@ -86,22 +89,23 @@ defmodule EctoShorts.Actions do
       iex> "can't be blank" in errors_on(changeset).first_name
       true
   """
-  def create(schema, params), do: Repo.call(:insert, [schema.create_changeset(params)])
+  def create(schema, params), do: Repo.insert(schema.create_changeset(params))
 
-  @spec find_or_create(Ecto.Schema.t, map) :: {:ok, Ecto.Schema.t} | {:error, Ecto.Changeset.t}
+  @spec find_or_create(Ecto.Query.t(), map) ::
+          {:ok, Ecto.Schema.t()} | {:error, Ecto.Changeset.t()}
   @doc "Finds a schema by params or creates one if it isn't found"
-  def find_or_create(schema, params) do
-    with {:error, %{code: :not_found}} <- find(schema, params) do
+  def find_or_create(query, params) do
+    with {:error, _} <- find(query, params) do
```

Comment:
Not sure if this was the right move here. It didn't want to pattern match according to dialyzer. Are there other error codes?